src/foundry/client/pixi/webgl/shaders/lighting/base-lighting.d.mts

```diff
@@ -0,0 +1,182 @@
+export {};
+declare global {
+  interface ShaderTechnique {
+    /** The numeric identifier of the technique */
+    id: number;
+
+    /** The localization string that labels the technique */
+    label: string;
+
+    /**  The coloration shader fragment when the technique is used */
+    coloration: string | undefined;
+
+    /** The illumination shader fragment when the technique is used */
+    illumination: string | undefined;
+
+    /** The background shader fragment when the technique is used */
+    background: string | undefined;
```

Comment:
You probably want `?` as well, if an explicit undefined is okay it's likely an implicit one is okay too.

Reply:
This was copying the foundry type exactly (it does not mark them as optional), but yes, this seems accurate. Done and noted in the documentation issue.